packages/react-core/src/components/Menu/examples/Menu.md

```diff
@@ -113,54 +113,88 @@ class MenuIconsList extends React.Component {
 
 ### With flyout
 
+The flyout will automatically position to the left or top if it would otherwise go outside the window. The menu must be placed in a container outside the main content like [Popover](/components/popover) or [Tooltip](/components/tooltip) since it may go over the side nav.
+
 ```js
 import React from 'react';
-import { Menu, MenuContent, MenuList, MenuItem } from '@patternfly/react-core';
-
-class MenuWithFlyout extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      activeItem: 0
-    };
-    this.onSelect = (event, itemId) => {
-      this.setState({
-        activeItem: itemId
-      });
-    };
+import { Menu, MenuContent, MenuList, MenuItem, Dropdown, DropdownToggle } from '@patternfly/react-core';
+
+MenuWithFlyout = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const menuRef = React.useRef();
+  const numFlyouts = 15;
+  const FlyoutMenu = ({ depth, children }) => (
+    <Menu key={depth} containsFlyout id={`menu-${depth}`} onSelect={onSelect}>
+      <MenuList>
+        <MenuItem aria-label="Has flyout menu" flyoutMenu={children} itemId={`next-menu-${depth}`}>Next menu</MenuItem>
+        {[...Array(numFlyouts - depth).keys()].map(j =>
+          <MenuItem key={`${depth}-${j}`} itemId={`${depth}-${j}`}>Menu {depth} item {j}</MenuItem>
+        )}
+        <MenuItem aria-label="Has flyout menu" flyoutMenu={children} itemId={`next-menu-2-${depth}`}>Next menu</MenuItem>
+      </MenuList>
+    </Menu>
+  );
+  let curFlyout = <FlyoutMenu depth={1} />;
+  for (let i = 2; i < numFlyouts - 1; i++) {
+    curFlyout = <FlyoutMenu depth={i}>{curFlyout}</FlyoutMenu>;
   }
 
-  render() {
-    const { activeItem } = this.state;
-    const flyoutMenu = (
-      <Menu onSelect={this.onSelect} activeItemId={activeItem}>
-        <MenuContent>
-          <MenuList>
-            <MenuItem itemId={10}>Application Grouping</MenuItem>
-            <MenuItem itemId={11}>Count</MenuItem>
-            <MenuItem itemId={12}>Labels</MenuItem>
-            <MenuItem itemId={13}>Annotations</MenuItem>
-          </MenuList>
-        </MenuContent>
-      </Menu>
-    );
+  // On keypress auto focus first element.
+  const id = 'root-menu';
+  const onEnter = event => {
+    if (!isOpen) {
+      setTimeout(() => {
+        const menu = document.getElementById(id);
+        if (menu) {
+          const firstFocusable = menu.querySelector('ul > li > button[tabindex="0"]');
+          if (firstFocusable) {
+            firstFocusable.focus();
+          }
+        }
+      }, 0);
+    }
+  };
 
-    return (
-      <Menu containsFlyout onSelect={this.onSelect} activeItemId={activeItem}>
+  const onSelect = (_ev, itemId) => {
+    console.log('select', itemId);
+    if (!itemId.includes('next-menu')) {
+      setIsOpen(false);
+    }
+  };
+
+  return (
+    <Dropdown
```

Comment:
Would it be worth following our composable menu demos (using MenuToggle & Popper) instead of using Dropdown, and moving this example to the composable menu page? The other Menu examples don't use a toggle so this example looks slightly out of place

Reply:
I was unaware of that section and I agree flyout fits better there and should use MenuToggle & Popper instead of `<Dropdown>`. I'll address that in my next commit.